### PR TITLE
[ADAM-1057] Remove workaround for gzip/BGZF compressed VCF headers

### LINKED ISSUE #1057: Remove workaround for gzip/BGZF compressed VCF headers
Remove the workaround here

https://github.com/bigdatagenomics/adam/blob/master/adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala#L159

for gzip/BGZF compressed VCF headers.  The issue has been fixed upstream

Fix VCFHeaderReader for gzipped and BGZF VCF formats
https://github.com/HadoopGenomics/Hadoop-BAM/pull/97

Note this will require bumping the dependency version of Hadoop-BAM to 7.5.1 (or 7.6.0) which may bring along a transitive dependency version bump of htsjdk to 2.5.0, see https://github.com/HadoopGenomics/Hadoop-BAM/pull/107.

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #1057 

Requires https://github.com/HadoopGenomics/Hadoop-BAM/pull/106 to be merged and deployed as version 7.5.1-SNAPSHOT.  Will need to be updated to 7.6.0 when that release is made.

### CODE REVIEW

**Comment by heuermh** [comment]
Jenkins, test this please.

Issue https://github.com/HadoopGenomics/Hadoop-BAM/pull/106 has been closed and a new shanpshot should have been deployed.

**Comment by heuermh** [comment]
Jenkins, retest this please.

File: pom.xml [hunk 1]
```diff
@@ -25,12 +25,12 @@
     <parquet.version>1.8.1</parquet.version>
     <!-- Edit the following line to configure the Hadoop (HDFS) version. -->
     <hadoop.version>2.6.0</hadoop.version>
-    <hadoop-bam.version>7.5.0</hadoop-bam.version>
+    <hadoop-bam.version>7.5.1-SNAPSHOT</hadoop-bam.version>
```
Comment (fnothaft) [inline pom.xml]: This will go to 7.5.1 when that pushes to Maven Central, right?

Comment (heuermh) [inline pom.xml]: 7.6.0 was just released and has made it to Maven Central, so I'll push & rebase with the new version shortly

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala [hunk 2]
```diff
@@ -59,7 +59,7 @@ import org.bdgenomics.utils.io.LocalFileByteAccess
 import org.bdgenomics.utils.misc.HadoopUtil
 import org.bdgenomics.utils.misc.Logging
 import org.seqdoop.hadoop_bam._
-import org.seqdoop.hadoop_bam.util.{ BGZFCodec, SAMHeaderReader, VCFHeaderReader, WrapSeekable }
+import org.seqdoop.hadoop_bam.util.{ BGZFCodec, BGZFEnhancedGzipCodec, SAMHeaderReader, VCFHeaderReader, WrapSeekable }
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala]: Small nit for readability: can you change this to either `org.seqdoop.hadoop_bam.util._` or split it up across multiple lines? I think we typically go to importing the whole package at 4 imports.

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala [hunk 3]
```diff
@@ -149,30 +149,9 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
...
-            v._2
-              .get
-              .asInstanceOf[VariantContextWithHeader]
-              .getHeader
-          }).first
-
-        headerToMetadata(vcfHeader)
-      }
-    }
+    val vcfHeader = VCFHeaderReader.readHeaderFrom(WrapSeekable.openPath(sc.hadoopConfiguration,
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala]: It feels great to get rid of that code above here!

**Comment by fnothaft** [comment]
LGTM, modulo one import nit. Let me know when you've updated it, and I can merge. I'm looking forward to having this in, @heuermh!

**Comment by fnothaft** [comment]
Hmm, I'm having bad times running this on the cluster on the 1KG .vcf.gz files. Unfortunately, I'm getting an uninformative error—the exception isn't serializable, which throws yet another exception—let me try to debug that and get a useful error.

**Comment by heuermh** [comment]
Rebased!

**Comment by jpdna** [comment]
+1 This works fine for me on a compressed vcf
I'll merge tomorrow unless someone suggests otherwise

**Comment by jpdna** [comment]
Merged 308cc586fb28367df3
Thanks for the PR @heuermh!

**Pull request closed without merging**